jdaviz/core/helpers.py

```diff
@@ -402,6 +405,72 @@ def show_in_new_tab(self, title=None):  # pragma: no cover
                       DeprecationWarning)
         return self.show(loc="sidecar:tab-after", title=title)
 
+    def get_data(self, data_label, cls='default', subsets_to_apply=None, statistic=None):
+        """
+        Returns data with name equal to `data_label` of type `cls` with subsets applied from
+        `subsets_to_apply`.
+
+        Parameters
+        ----------
+        cls : `~specutils.Spectrum1D`, 'default'
+            The type that data will be returned as.
+        data_label : str
+            Provide a label to retrieve a specific data set from data_collection.
+        subsets_to_apply : str, list, optional
+            Subsets that are to be applied to data before it is returned. If a string
+            is given, it will be turned into a list and looped through. A future implementation
+            will allow applying multiple subsets to data before it is returned.
```

Comment:
maybe we should just say string or None for now (and raise an error for other types) and just create a follow-up ticket.  Promising future support in the docstring could get us in trouble if we ever decide not to do that (or take another approach).